### DISCUSSION TITLE
Large SSD1320Z*2 Based OLED Help Needed

### DISCUSSION THREAD

**Discussion by buildxyz-git** [olikraus/u8g2#1396] [Q&A]
I'm working on a project with multiple OLED displays and so far every display I have purchased was supported by u8g2 and had an available constructor and works like magic.

I just received one of [these](https://www.aliexpress.com/item/1005001446494204.html?spm=a2g0o.productlist.0.0.33d359c8SnuX0W&algo_pvid=ff8976f3-cc3c-40ba-86db-907d0e797c30&algo_expid=ff8976f3-cc3c-40ba-86db-907d0e797c30-4&btsid=0b0a556716123996463648264e0704&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_) which is  large 3.83" OLED and want to get it up and running with the u8g2 library. The product page states the controller(s) are "**SSD1320Z*2**". I do see constructors for various SSD132X controllers but not at this resolution (320x132 ).

Can anyone explain the dual controller scenario? I can't seem to locate the exact data sheet. The display header does have two Chip Select line inputs (CS1 and CS2)

My knowledge of the controllers is very limited (like zero).

Any help is great!

![Screen Shot 2021-02-03 at 4 51 14 PM](https://user-images.githubusercontent.com/16586332/106828958-0f907e80-6640-11eb-9aa8-b6b8410ddc57.png)
![Screen Shot 2021-02-03 at 4 51 03 PM](https://user-images.githubusercontent.com/16586332/106828966-13240580-6640-11eb-815e-eb640060bd4d.png)

**Comment by JamesGKent** [comment]
my guess is that this is essentially two screens with separate controllers. likely split into two halves vertically. AFAIK u8g2 doesn't have any constructors that support dual controllers like this, so you would have to write your code to work on each half as its own screen.

**Reply by buildxyz-git** [reply]
Thanks for the input @JamesGKent! 
I had also thought this and I'm thinking this is probably the case now. It probably needs to be treated as 2 separate 160x132 screens. The SSD1320Z in undocumented AFAIK, though the [SSD1320 ](https://cdn.sparkfun.com/assets/d/b/1/f/d/SSD1320_1.0__to_Wisechip_.pdf)has an available data sheet, I'm really not sure of the differences between the two controllers. 
The SSD1320 is currently not supported by u8g2 though I did see [an issue](https://github.com/olikraus/u8g2/issues/1351) raised to support that controller which didn't go anywhere.
I did try a bunch of the constructors and the U8G2_SSD1326_ER_256X32_2_4W_HW_SPI did produce some dim junk on the screen when connected to CS2:

![IMG_4712](https://user-images.githubusercontent.com/16586332/107069569-0108ab00-6797-11eb-8c17-f051f61ae7e8.jpg)

**Comment by nikitalita** [comment]
@buildxyz-git I see you got this to work in your pinball project, albeit with noticeable display flickering; is this an inherent limitation of the device?

**Reply by buildxyz-git** [reply]
It works great. The flickering is probably due to the shutter speed of the camera.

**Reply by nikitalita** [reply]
Duh, of course. How is the refresh rate? Are you able to update at 60hz?

**Reply by buildxyz-git** [reply]
It's decent. I updated it @30hz and it's acceptable in my opinion. I'm currently clocking the SPI interface @10MHz.

**Reply by nikitalita** [reply]
Thank you! Do you plan to publish the source code for the display?

**Reply by buildxyz-git** [reply]
No plans for that. My code is very particular to my application and @olikraus has already pushed the underlying SSD1320 driver which is required to use it like any other OLED display.
